### PR TITLE
ci: Return custom exit code to indicate known shm leak failure in L0_backend_python bls test

### PR DESCRIPTION
**Pull request by krishung5**
#### What does the PR do?

Return custom exit code `123` to indicate known shm leak failure in L0_backend_python bls test. CI will catch the custom exit code and allow the L0_backend_python test to fail if the known issues occur.

In order to propagate the exit code correctly, Python unittest and Pytest can not be used at the same time. In this PR, the `python_unittest.py` is replaced with `test_infer_shm_leak.py` which uses Pytest directly.

#### Checklist
- [x] PR title reflects the change and is of format `<commit_type>: <Title>`
- [x] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [ ] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [x] ci
- [ ] docs
- [ ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [x] test

#### Related PRs:

tritonserver MR !236.

#### Test plan:

L0_backend_python should fail with `allowed to fail` if it's failing due to the known shm leak from bls sub test.

- CI Pipeline ID: 17025308

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
- closes GitHub issue: #xxx

### CODE REVIEW

File: qa/L0_backend_python/bls/test.sh [hunk 1]
```diff
@@ -119,30 +119,35 @@ if [[ ${TEST_WINDOWS} == 0 ]]; then
...
+                    cat $SERVER_LOG
+                    RET=1
+                fi
             fi
 
-            if [[ $CUDA_MEMORY_POOL_SIZE_MB -eq 128 ]]; then
+            if [[ $CUDA_MEMORY_POOL_SIZE_MB -eq 256 ]]; then
                 if [ `grep -c "Failed to allocate memory from CUDA memory pool" $SERVER_LOG` != "0" ]; then
-                    echo -e "\n***\n*** Expected to use CUDA memory pool for all tests when CUDA_MEMOY_POOL_SIZE_MB is 128 MB for 'bls' $BLS_KIND test\n***"
+                    echo -e "\n***\n*** Expected to use CUDA memory pool for all tests when CUDA_MEMOY_POOL_SIZE_MB is 256 MB for 'bls' $BLS_KIND test\n***"
```
Comment (rmccorm4) [inline qa/L0_backend_python/bls/test.sh:150]:
```suggestion
                    echo -e "\n***\n*** Expected to use CUDA memory pool for all tests when CUDA_MEMORY_POOL_SIZE_MB is 256 MB for 'bls' $BLS_KIND test\n***"
```

File: qa/L0_backend_python/setup_python_enviroment.sh [hunk 2]
```diff
@@ -151,7 +151,7 @@ apt-get update && apt-get -y install \
                             libboost-dev
 rm -f /usr/bin/python3 && \
 ln -s "/usr/bin/python3.${PYTHON_ENV_VERSION}" /usr/bin/python3
-pip3 install --upgrade install requests numpy virtualenv protobuf
```
Comment (rmccorm4) [inline qa/L0_backend_python/setup_python_enviroment.sh:154]: Nice find :rocket:

File: qa/common/shm_util.py [hunk 3]
```diff
@@ -441,6 +441,9 @@ def __exit__(self, type, value, traceback):
                     print(
                         f"Shared memory leak detected [{shm_region}]: {curr_shm_free_size} (curr free) < {prev_shm_free_size} (prev free)."
                     )
+                    # Known shared memory leak of 480 bytes in BLS test.
```
Comment (rmccorm4) [inline qa/common/shm_util.py:444]: Do we have any ticket to eventually find/resolve this issue all together? Would be good to leave the FIXME `DLIS-XXXX` here if so.

**Review by rmccorm4 — COMMENTED** [review]
LGTM, just a minor comment on ticket reference

File: qa/common/shm_util.py (hunk 3 above)
Comment (krishung5) [reply qa/common/shm_util.py]: Good call, added the ticket reference.

**Review by rmccorm4 — APPROVED** [review]

**Review by kthui — APPROVED** [review]

**Pull request merged**